tests/func/test_checkout.py

```diff
@@ -531,3 +532,8 @@ def test_should_not_relink_on_unchanged_dependency(link, dvc_repo, repo_dir):
         dvc_repo.checkout(repo_dir.DATA_DIR + Stage.STAGE_FILE_SUFFIX)
 
     assert not mock_link.called
+
+
+def test_should_raise_on_no_dir_cache(cloned_dvc_repo):
+    with pytest.raises(DirCacheLoadingError):
+        cloned_dvc_repo.checkout("data_dir.dvc")
```

Comment:
Using `cloned_dvc_repo` fixture feels really weird here. This test is not even really related to external repos. What you are testing here is that `dvc checkout` will raise an exception if it doesn't have cache for a directory and is not able to download *.dir cache from default remote. I'm pretty sure you could do that with:
```
def test...(repo_dir, dvc_repo):
    stage, = dvc_repo.add(repo_dir.DATA_DIR)
    with pytest.raises(DirCacheLoadingError):
        dvc_repo.checkout(stage.path)
```

But the actuall question here is why are we raising at all? This issue is recoverable and we should simply say that we don't have cache for `data_dir` and it won't be created, as we always do in such cases.

Reply:
Hmm, I don't quite get how it is recoverable. We want to check out some target, there is no cache, so we cannot do that, that does not seem to be recoverable?

Reply:
@pared Same as we do right now: if there is no cache: don't create the file. E.g. you only have data sources in your cache to save space and now you need to checkout them and then `dvc repro` to get all end results and intermediate results.

Reply:
Ok, but current behaviour is the sole reason of many errors flooding sterr, described in #2464. If we let checkout process further (after failing to get all files number), it will still fail on actual checkout.

Reply:
@pared There are lot of ugly messages, which should be fixed or handled more gracefully, but the behaviour of no cache - no file should stay. Though, we might want to treat it like we do pull/push: checkout everything we can and only then error-out. But it seems like the ticket itself is mostly about those confusing messages.

Reply:
Ok, that makes sense. But then to solve original issue and make messages less confusing, I need to somewhat know that download failed, or supress log when obtaining file number.

Reply:
For the record: discussed privately and decided to use `self.cache.open()` instead of `download` in `load_dir_cache`.